server/channels/web/web_test.go

```diff
@@ -31,6 +31,8 @@ var apiClient *model.Client4
 var URL string
 
 type TestHelper struct {
+	t testing.TB
```

Comment:
Would it be possible to avoid this pattern in favour of passing `testing.TB` to any methods that need it?

The challenge I've run up against is that this value isn't "right" for subtests that use a common test helper e.g.:

```go
func TestThing(t * testing.T) {
    th := testHelper{t}

    t.Run("subtest", func(t *testing.T) {
        // th.DoSomething will use the outer `*testing.T`, not the inner one.
        th.DoSomething()
    })
}
```

It still "works", but any assertions can be tied to the outer test and create confusing results. In cases where I've really needed to remember the current test context, I've added a `Reset(t)` helper method, but I'd wager we could just pass `*testing.T` instead.

Thoughts?

Reply:
Thanks for the careful review and the great observation. Placing a `testing.TB` in a struct indeed causes issues. I ended up not needing a `Reset(t)` method. Passing `testing.TB` was enough.

Much cleaner now!